enterprise_subsidy/apps/api/v1/tests/test_views.py

```diff
@@ -814,6 +812,31 @@ def test_create_missing_inputs(self, missing_post_arg):
         # Just make sure there's any parseable json which is likely to contain an explanation of the error.
         assert response.json()
 
+    def test_fetch_transaction_with_external_reference(self):
+        """
+        Test fetching a Transaction with external reference.
+        """
+        url = reverse("api:v1:transaction-list")
+        # Create privileged staff user that should be able to create Transactions.
+        self.set_up_operator()
+        external_reference_id = "foobar"
+        external_transaction_reference = ExternalTransactionReferenceFactory(
+            external_reference_id=external_reference_id,
+            external_reference_type=TransactionReferenceTypeChoices.GEAG_ALLOCATION_ID,
+        )
+        transaction_with_external_reference = TransactionFactory(
+            quantity=-1000,
+            ledger=self.subsidy_1.ledger,
+            lms_user_id=STATIC_LMS_USER_ID,  # This is the only transaction belonging to the requester.
+            subsidy_access_policy_uuid=self.subsidy_access_policy_1_uuid,
+            content_key=self.content_key_1,
+        )
```

Comment:
Transaction should be created before an ExternalTransactionReference to avoid creating an orphaned Transaction.